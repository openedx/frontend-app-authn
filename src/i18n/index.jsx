import { messages as paragonMessages } from '@edx/paragon';

import arMessages from './messages/ar.json';
import deMessages from './messages/de.json';
import dedeCAMessages from './messages/de_DE.json';
import es419Messages from './messages/es_419.json';
import frMessages from './messages/fr.json';
import frCAMessages from './messages/fr_CA.json';
import hiMessages from './messages/hi.json';
import itMessages from './messages/it.json';
import ititCAMessages from './messages/it_IT.json';
import ptMessages from './messages/pt.json';
import ptptCAMessages from './messages/pt_PT.json';
import ruMessages from './messages/ru.json';
import ukMessages from './messages/uk.json';
import zhcnMessages from './messages/zh_CN.json';
// no need to import en messages-- they are in the defaultMessage field

const appMessages = {
  ar: arMessages,
  'es-419': es419Messages,
  fr: frMessages,
  'zh-cn': zhcnMessages,
  pt: ptMessages,
  it: itMessages,
  de: deMessages,
  hi: hiMessages,
  'fr-ca': frCAMessages,
  ru: ruMessages,
  uk: ukMessages,
  'de-de': dedeCAMessages,
  'it-it': ititCAMessages,
  'pt-pt': ptptCAMessages,
};

export default [
  paragonMessages,
  appMessages,
];
