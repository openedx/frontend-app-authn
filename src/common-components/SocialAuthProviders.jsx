import React from 'react';
import PropTypes from 'prop-types';

import { getConfig } from '@edx/frontend-platform';
import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSignInAlt } from '@fortawesome/free-solid-svg-icons';

import messages from './messages';
import { LOGIN_PAGE, SUPPORTED_ICON_CLASSES } from '../data/constants';

function SocialAuthProviders(props) {
  const { intl, referrer, socialAuthProviders } = props;

  function handleSubmit(e) {
    e.preventDefault();

    const url = e.currentTarget.dataset.providerUrl;
    window.location.href = getConfig().LMS_BASE_URL + url;
  }

  const socialAuth = socialAuthProviders.map((provider, index) => (
    <button
      id={provider.id}
      key={provider.id}
      type="button"
      className={`btn-social btn-${provider.id} ${index % 2 === 0 ? 'mr-3' : ''}`}
      data-provider-url={referrer === LOGIN_PAGE ? provider.loginUrl : provider.registerUrl}
      onClick={handleSubmit}
    >
      {provider.iconImage ? (
        <div aria-hidden="true">
          <img className="icon-image" src={provider.iconImage} alt={`icon ${provider.name}`} />
        </div>
      )
        : (
          <>
            <div className="font-container" aria-hidden="true">
              <FontAwesomeIcon
                icon={SUPPORTED_ICON_CLASSES.includes(provider.iconClass) ? ['fab', provider.iconClass] : faSignInAlt}
              />
            </div>
          </>
        )}
      <span id="provider-name" className="mr-auto pl-2" aria-hidden="true">{provider.name}</span>
      <span className="sr-only">
        {referrer === LOGIN_PAGE
          ? intl.formatMessage(messages['sso.sign.in.with'], { providerName: provider.name })
          : intl.formatMessage(messages['sso.create.account.using'], { providerName: provider.name })}
      </span>
    </button>
  ));

  return <>{socialAuth}</>;
}

SocialAuthProviders.defaultProps = {
  referrer: LOGIN_PAGE,
  socialAuthProviders: [],
};

SocialAuthProviders.propTypes = {
  intl: intlShape.isRequired,
  referrer: PropTypes.string,
  socialAuthProviders: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string,
    name: PropTypes.string,
    iconClass: PropTypes.string,
    iconImage: PropTypes.string,
    loginUrl: PropTypes.string,
    registerUrl: PropTypes.string,
  })),
};

export default injectIntl(SocialAuthProviders);
