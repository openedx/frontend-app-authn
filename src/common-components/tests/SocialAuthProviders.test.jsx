import React from 'react';

import { IntlProvider } from '@edx/frontend-platform/i18n';
import renderer from 'react-test-renderer';

import registerIcons from '../RegisterFaIcons';
import SocialAuthProviders from '../SocialAuthProviders';

registerIcons();

describe('SocialAuthProviders', () => {
  let props = {};

  const appleProvider = {
    id: 'oa2-apple-id',
    name: 'Apple',
    iconClass: null,
    iconImage: 'https://edx.devstack.lms/logo.png',
    loginUrl: '/auth/login/apple-id/?auth_entry=login&next=/dashboard',
  };

  const facebookProvider = {
    id: 'oa2-facebook',
    name: 'Facebook',
    iconClass: null,
    iconImage: 'https://edx.devstack.lms/facebook-logo.png',
    loginUrl: '/auth/login/facebook/?auth_entry=login&next=/dashboard',
  };

  it('should match social auth provider with iconImage snapshot', () => {
    props = { socialAuthProviders: [appleProvider, facebookProvider] };

    const tree = renderer.create(
      <IntlProvider locale="en">
        <SocialAuthProviders {...props} />
      </IntlProvider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should match social auth provider with iconClass snapshot', () => {
    props = {
      socialAuthProviders: [{
        ...appleProvider,
        iconClass: 'google',
        iconImage: null,
      }],
    };

    const tree = renderer.create(
      <IntlProvider locale="en">
        <SocialAuthProviders {...props} />
      </IntlProvider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should match social auth provider with default icon snapshot', () => {
    props = {
      socialAuthProviders: [{
        ...appleProvider,
        iconClass: 'default',
        iconImage: null,
      }],
    };

    const tree = renderer.create(
      <IntlProvider locale="en">
        <SocialAuthProviders {...props} />
      </IntlProvider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
