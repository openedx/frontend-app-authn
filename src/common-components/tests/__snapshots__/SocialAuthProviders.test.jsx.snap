// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialAuthProviders should match social auth provider with default icon snapshot 1`] = `
<button
  className="btn-social btn-oa2-apple-id mr-3"
  data-provider-url="/auth/login/apple-id/?auth_entry=login&next=/dashboard"
  id="oa2-apple-id"
  onClick={[Function]}
  type="button"
>
  <div
    aria-hidden="true"
    className="font-container"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-sign-in-alt fa-w-16 "
      data-icon="sign-in-alt"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M416 448h-84c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h84c17.7 0 32-14.3 32-32V160c0-17.7-14.3-32-32-32h-84c-6.6 0-12-5.4-12-12V76c0-6.6 5.4-12 12-12h84c53 0 96 43 96 96v192c0 53-43 96-96 96zm-47-201L201 79c-15-15-41-4.5-41 17v96H24c-13.3 0-24 10.7-24 24v96c0 13.3 10.7 24 24 24h136v96c0 21.5 26 32 41 17l168-168c9.3-9.4 9.3-24.6 0-34z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </div>
  <span
    aria-hidden="true"
    className="mr-auto pl-2"
    id="provider-name"
  >
    Apple
  </span>
  <span
    className="sr-only"
  >
    Sign in with Apple
  </span>
</button>
`;

exports[`SocialAuthProviders should match social auth provider with iconClass snapshot 1`] = `
<button
  className="btn-social btn-oa2-apple-id mr-3"
  data-provider-url="/auth/login/apple-id/?auth_entry=login&next=/dashboard"
  id="oa2-apple-id"
  onClick={[Function]}
  type="button"
>
  <div
    aria-hidden="true"
    className="font-container"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-google fa-w-16 "
      data-icon="google"
      data-prefix="fab"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 488 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M488 261.8C488 403.3 391.1 504 248 504 110.8 504 0 393.2 0 256S110.8 8 248 8c66.8 0 123 24.5 166.3 64.9l-67.5 64.9C258.5 52.6 94.3 116.6 94.3 256c0 86.5 69.1 156.6 153.7 156.6 98.2 0 135-70.4 140.8-106.9H248v-85.3h236.1c2.3 12.7 3.9 24.9 3.9 41.4z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </div>
  <span
    aria-hidden="true"
    className="mr-auto pl-2"
    id="provider-name"
  >
    Apple
  </span>
  <span
    className="sr-only"
  >
    Sign in with Apple
  </span>
</button>
`;

exports[`SocialAuthProviders should match social auth provider with iconImage snapshot 1`] = `
Array [
  <button
    className="btn-social btn-oa2-apple-id mr-3"
    data-provider-url="/auth/login/apple-id/?auth_entry=login&next=/dashboard"
    id="oa2-apple-id"
    onClick={[Function]}
    type="button"
  >
    <div
      aria-hidden="true"
    >
      <img
        alt="icon Apple"
        className="icon-image"
        src="https://edx.devstack.lms/logo.png"
      />
    </div>
    <span
      aria-hidden="true"
      className="mr-auto pl-2"
      id="provider-name"
    >
      Apple
    </span>
    <span
      className="sr-only"
    >
      Sign in with Apple
    </span>
  </button>,
  <button
    className="btn-social btn-oa2-facebook "
    data-provider-url="/auth/login/facebook/?auth_entry=login&next=/dashboard"
    id="oa2-facebook"
    onClick={[Function]}
    type="button"
  >
    <div
      aria-hidden="true"
    >
      <img
        alt="icon Facebook"
        className="icon-image"
        src="https://edx.devstack.lms/facebook-logo.png"
      />
    </div>
    <span
      aria-hidden="true"
      className="mr-auto pl-2"
      id="provider-name"
    >
      Facebook
    </span>
    <span
      className="sr-only"
    >
      Sign in with Facebook
    </span>
  </button>,
]
`;
